try.rb

```diff
@@ -662,9 +662,31 @@ def handle_delete(try_dir)
 
     if confirmation == "YES"
       begin
+        # Check if current working directory is within the path being deleted
+        current_dir = Dir.pwd rescue nil
+        deleted_path = File.realpath(try_dir[:path]) rescue try_dir[:path]
+        is_current_dir_affected = current_dir && 
+          (current_dir == deleted_path || current_dir.start_with?(deleted_path + "/"))
+        
         FileUtils.rm_rf(try_dir[:path])
         @delete_status = "Deleted: #{try_dir[:basename]}"
         @all_tries = nil  # Clear cache to reload tries
+        
+        # If current directory was deleted, change to a safe location
+        if is_current_dir_affected
```

Comment:
I feel this entire thing could be done vastly simpler by just appending an cd .. to the end of the script if the current directory is the deleted one.